src/ruby/puppet-server-lib/puppet/server/network/http/handler.rb

```diff
@@ -32,7 +34,12 @@ def path(request)
   end
 
   def body(request)
-    request["body"]
+    body = request["body"]
+    if body.java_kind_of?(InputStream)
+      body.to_io.read()
```

Comment:
Are we concerned about `IO#read` copying the entire file content in memory?  As this pull request is concerned with file bucket files, do we have a sense of the behavior when large files are being stored?